inc/TRestRawSignalEvent.h

```diff
@@ -71,11 +72,22 @@ class TRestRawSignalEvent : public TRestEvent {
         for (int n = 0; n < GetNumberOfSignals(); n++) fSignal[n].SetTailPoints(p);
     }
 
-    void SetBaseLineRange(TVector2 blRange) { SetBaseLineRange(blRange.X(), blRange.Y()); }
+    void SetBaseLineRange(TVector2 blRange, std::string option = "") {
+        SetBaseLineRange(blRange.X(), blRange.Y(), option);
+    }
+
+    void SetBaseLineRange(Int_t from, Int_t to, std::string option = "") {
+        fBaseLineRange = TVector2(from, to);
+        for (int n = 0; n < GetNumberOfSignals(); n++) fSignal[n].CalculateBaseLine(from, to, option);
+    }
+
+    /// Uses for the baseline calculation not the standard deviation but IQR (interquartile range) divided
+    /// by 1.349
+    void SetBaseLineRangeRobust(TVector2 blRange) { SetBaseLineRangeRobust(blRange.X(), blRange.Y()); }
 
-    void SetBaseLineRange(Int_t from, Int_t to) {
+    void SetBaseLineRangeRobust(Int_t from, Int_t to) {
```

Comment:
In reality, this method should not be renamed, it should be kept as `SetBaseLineRange` since the main goal of this method is to initialize `fBaseLineRange` metadata member. However, when we initialize `fBaseLineRange` metadata member we must recalculate the baseline average and fluctuation. So the appropriate thing would be to keep `SetBaseLineRange` method name, and just call `CalculateBaseLineAerage` or `CalculateBaseLineMedian`. `CalculateBaseLineMedian` is the name I think it should have `CalculateBaseLineRobust`.

So, probably you have 2 options.
- Provide a third optional argument where you specify the method for baseline calculation, that you already somewhere else.
- Define a new metadata member at `TRestRawSignalEvent` that tells the options used at the `TRestRawSignal` event processing.  This could be done in diferent ways, but I would probably chose to use an integer number, where the first digit of the units tells us which baseline calculation method we use (average/method), the second digit tells us which is the method for baseline fluctuations (RMS/IQR), etc. More options might appear in the future.

We would then have a data member:

```
Int_t fOptions = 11;
```

// For example this function would return true if the tens would be equal to 2.
Bool_t IsCalculateBaseLineMedian() { return (fOptions/10)%10 == 2; }

// For example this function would return true if the tens would be equal to 1.
Bool_t IsCalculateBaseLineAverage() { return (fOptions/10)%10 == 1; }

// Then you implement a public method (previous ones are private).
void SetOptions ( Int_t options ) { fOptions = options; }

// You could also implement other methods to make it more friendly, but documentation should be good enough to understand the options available and the number to be used.

```

The inside the code you could just select the methods internally

```
SetBaseLineRange(from,to) {
if (IsCalculateBaseLineAverage() ) CalculateBaseLineAverage(from,to);
if (IsCalculateBaseLineMedian() ) CalculateBaseLineMedian(from,to);
}

Reply:
We could discuss this online

Reply:
I changed and renamed several methods to make a bit cleare what each is doing:

- To calculate the baseline average: `CalculateBaseLineMean` and `CalculateBaseLineMedian`, which both write `fBaseLine`
- To calculate the baseline fluctuations: `CalculateBaseLineSigmaSD` and `CalculateBaseLineSigmaIQR`, which both write `fBaseLineSigma`
- `TRestRawSignal::CalculateBaseLine` and `TRestRawSignalEvent::SetBaseLineRange` both call the methods above and accept a third option "ROBUST" (after the signal range), which switches from Mean/SD to Median/IQR.

I kept the option as "ROBUST", because I don't find a better way to describe the switch to Median/IQR. In the documentation I think I sufficiently explained, what exactly it is doing.

Another metadata member to see which option was used I didn't include yet, because so far the "ROBUST" option is only used by `TRestRawVetoAnalysisProcess`, where it is hardcoded.

Reply:
>> Another metadata member to see which option was used I didn't include yet, because so far the "ROBUST" option is only used by TRestRawVetoAnalysisProcess, where it is hardcoded.

Ok but I think we should add the option also at `TRestRawSignalAnaysisProcess`.

On the other hand, it is also interesting that it is not used at many places, so that the change does not need yet to be applied at many places.